bundles/org.openhab.core.io.monitor/src/main/java/org/openhab/core/io/monitor/internal/metrics/ThingStateMetric.java

```diff
@@ -111,10 +113,13 @@ public Set<String> getSubscribedEventTypes() {
 
     @Override
     public void receive(Event event) {
-        logger.trace("Received ThingStatusInfo(Changed)Event...");
-        String thingId = event.getTopic().substring(THINGSTATUS_TOPIC_PREFIX.length(),
-                event.getTopic().lastIndexOf('/'));
-        ThingStatus status = gson.fromJson(event.getPayload(), ThingStatusInfo.class).getStatus();
-        createOrUpdateMetricForBundleState(thingId, status.ordinal());
+        if (event instanceof ThingStatusInfoEvent) {
+            logger.trace("Received ThingStatusInfo(Changed)Event...");
+            String thingUid = ((ThingStatusInfoEvent) event).getThingUID().getAsString();
```

Comment:
A more direct way of getting at the thingUid, rather than parsing it as a string out of the event topic. Since we are only subscribing for events of this type, we are safe to cast to the event type, after checking that something horrible wrong did not happen of course.